content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade.md

```diff
@@ -4,32 +4,34 @@ reviewers:
 title: Upgrading kubeadm clusters
 content_type: task
 weight: 20
-min-kubernetes-server-version: 1.19
+min-kubernetes-server-version: "{{< skew latestVersion >}}"
 ---
 
 <!-- overview -->
 
 This page explains how to upgrade a Kubernetes cluster created with kubeadm from version
-1.18.x to version 1.19.x, and from version 1.19.x to 1.19.y (where `y > x`).
+**{{< skew prevMinorVersion >}}** to version **{{< skew latestVersion >}}**, and from version
+**{{< skew latestVersion >}}.x** to **{{< skew latestVersion >}}.y** (where `y > x`).
 
 To see information about upgrading clusters created using older versions of kubeadm,
-please refer to following pages instead:
+copy the following link into your browser but replace `xx` with the desired MINOR version
+to upgrade to (for example `19`):
 
-- [Upgrading kubeadm cluster from 1.17 to 1.18](https://v1-18.docs.kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/)
-- [Upgrading kubeadm cluster from 1.16 to 1.17](https://v1-17.docs.kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/)
-- [Upgrading kubeadm cluster from 1.15 to 1.16](https://v1-16.docs.kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/)
-- [Upgrading kubeadm cluster from 1.14 to 1.15](https://v1-15.docs.kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-15/)
-- [Upgrading kubeadm cluster from 1.13 to 1.14](https://v1-15.docs.kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14/)
```

Comment:
The existing unordered list seems better for readers - a quick scan, one click, not much thinking.

Reply:
agreed that this degrades the UX a little

unless there is a way to automate the list links, i suggest we log an issue at k/website about it as one of the goals of the PR is to remove the kubeadm maintainer commitment for sending a kubeadm upgrade PR for the docs on every release.

Reply:
actually, i will try to construct the links in this PR.

for the versions in skew, we could use skew + replace for that:
https://github.com/kubernetes/website/blob/8e2544f1902d61795e62db1125d0d5d4a8157664/layouts/shortcodes/skew.html#L49-L51
https://gohugo.io/functions/replace/

so maybe something like:

```
- [Upgrading kubeadm cluster from {{< skew oldestMinorVersion >}} to {{< skew prevMinorVersion >}}](https://v{{ replace {{< skew prevMinorVersion >}} "." "-" }}.docs.kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/)
```

so if the current doc is 19->20 we can also have the link for 18->19 from skew, but not the links for the older versions
i guess we can try Hugo's math for that https://gohugo.io/functions/math/

Reply:
:+1: for logging an issue and dealing with that later.

I could have a go at a shortcode to repeat a link across all supported versions, if they all have the same path.

Reply:
@sftim i've added a new commit that adds `skew latestVersionAddMinor <number> "separator"`.
it makes it possible to produce any version with a custom separator.

LMK if an issue should still be logged.
we should note that the general logic in "skew.html" as is, will fail once k8s moves to v2.0 some day.